xds/src/main/java/io/grpc/xds/XdsLbState.java

```diff
@@ -85,26 +113,65 @@ final XdsComms shutdownAndReleaseXdsComms() {
     return xdsComms;
   }
 
-  static final class XdsComms {
-    private final ManagedChannel channel;
-    private final AdsStream adsStream;
+  private void maybeStartXdsComms() {
+    if (xdsComms != null) {
+      xdsComms = xdsComms.maybeRestart();
+      return;
+    }
 
-    XdsComms(ManagedChannel channel, AdsStream adsStream) {
-      this.channel = channel;
-      this.adsStream = adsStream;
+    // ** This is wrong **
+    // TODO: use name resolve to resolve addresses for balancerName, and create xdsComms in
+    // name resolver listener callback
+    ManagedChannel oobChannel = helper.createOobChannel(
+        new EquivalentAddressGroup(ImmutableList.<SocketAddress>of(new SocketAddress() {})),
```

Comment:
Maybe expose this fake EAG as a static final field visible to tests, and verify `createOobChannel()` with this EAG in tests. This way later when you pass real EAGs, those tests will fail and you can update them accordingly, so you are sure that you don't miss one.

Reply:
Added TODO because the current PR focus on fallback, and this method eventually will have the correct implementation and be tested.